python/ray/serve/_private/client.py

```diff
@@ -219,6 +220,59 @@ def _wait_for_deployment_deleted(self, name: str, timeout_s: int = 60):
         else:
             raise TimeoutError(f"Deployment {name} wasn't deleted after {timeout_s}s.")
 
+    def _wait_for_deployment_created(self, name: str, timeout_s: int = -1):
+        start = time.time()
+        while time.time() - start < timeout_s or timeout_s < 0:
+            status_bytes = ray.get(self._controller.get_deployment_status.remote(name))
+
+            if status_bytes is not None:
+                break
+
+            time.sleep(CLIENT_CHECK_CREATION_POLLING_INTERVAL_S)
+        else:
+            raise TimeoutError(
+                f"Deployment {name} did not become HEALTHY after {timeout_s}s."
+            )
+
+    def _wait_for_application_running(self, name: str, timeout_s: int = -1):
+        """Waits for the named application to enter "RUNNING" status.
+
+        Raises RuntimeError if the application enters the "DEPLOY_FAILED" status
+        instead.
+
+        Raises TimeoutError if this doesn't happen before timeout_s.
+        """
+        start = time.time()
+        while time.time() - start < timeout_s or timeout_s < 0:
```

Comment:
Same question as above, would we stuck in the loop forever and blocks other process?